starter-code/src/functions-and-arrays.js

```diff
@@ -1,25 +1,95 @@
 // Find the maximum
 
+function maxOfTwoNumbers (firstNumber, secondNumber){
+
+  if (firstNumber > secondNumber ){
+    return firstNumber;
+  }
+  else {
+    return secondNumber;
+  }
+}
+
 // Finding Longest Word
-var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
-];
+function findLongestWord(wordArray) {
+  
+
+
+  var longestWord = wordArray[0];
+  
+ for ( var i=0 ; i < wordArray.length; i++){
+    if (wordArray[i].length > longestWord.length) {  
+       longestWord = wordArray[i];
+    }
+   } 
+
+  return longestWord;
+}
+
+
+var words = ['mystery','brother', 'aviator',  'crocodile','pearl','orchard','crackpot'];
 
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray(arrayNumber) {
+
+  var result = 0;
+for ( var i=0 ; i < arrayNumber.length ; i++) {
+
+result = result + arrayNumber[i];
+
+}
+return result;
+}
+
+
+
 // Calculate the Average
 
+function averageNumbers(arrayNumber) {
+
+var result = 0;
+for ( var i=0 ; i < arrayNumber.length ; i++) {
+
```

Comment:
To keep your code dry (= avoid repetitions), you can also call the sumArray function that you created earlier.

``` javascript
function averageNumbers(arrayNumber) {
   var result = sumArray(arrayNumber);
   if (arrayNumber.length > 0) {
   var average = result / arrayNumber.length; 
   }
   return average;
 }
```